src/org/zaproxy/zap/extension/fuzz/httpfuzzer/processors/tagcreator/HttpFuzzerMessageProcessorTagCreator.java

```diff
@@ -0,0 +1,91 @@
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2015 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.fuzz.httpfuzzer.processors.tagcreator;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzResult;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerMessageProcessor;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerTaskProcessorUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class HttpFuzzerMessageProcessorTagCreator implements HttpFuzzerMessageProcessor {
+
+    public static final String NAME = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.name");
+    public static final String DESCRIPTION = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.desc");
+    public static final String TAG_CREATOR_STATE_KEY = "fuzz.HttpFuzzerMessageProcessorTagCreator.Tags";
+
+    private ITagRule _tagRule;
+
+    public HttpFuzzerMessageProcessorTagCreator(ITagRule tagRule){
+        _tagRule = tagRule;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public HttpMessage processMessage(HttpFuzzerTaskProcessorUtils utils, HttpMessage message) {
+        return message;
+    }
+
+    @Override
+    public boolean processResult(HttpFuzzerTaskProcessorUtils utils, HttpFuzzResult fuzzResult) {
+        List<String> tags = createTags(fuzzResult);
+        setTagsToCustomState(fuzzResult, tags);
+        return true;
+    }
+
+    private List<String> createTags(HttpFuzzResult fuzzResult) {
+        HttpMessage httpMessage = fuzzResult.getHttpMessage();
+        List<String> existingTags = getExistingTags(fuzzResult);
+        String responseMessage = getResponseMessage(httpMessage);
+        HttpResponseTagCreator tagCreator = new HttpResponseTagCreator(_tagRule, responseMessage, existingTags);
+        return tagCreator.Create();
+    }
+
+    private List<String> getExistingTags(HttpFuzzResult fuzzResult){
+        Map<String, Object> state =  fuzzResult.getCustomStates();
+        return getExistingTagsFromCustomState(state);
+    }
+
+    public static List<String> getExistingTagsFromCustomState(Map<String, Object> state){
+        List<String> existingTags  = new ArrayList<>();
```

Comment:
Consider using "Collections.emptyList()" instead. The "existingTags" is never changed, so in the case it's empty avoids creating a new ArrayList each time the method is called.